src/modules/logicModule/paymentClient/ERC20PaymentClient.sol

```diff
@@ -208,4 +184,53 @@ abstract contract ERC20PaymentClient is
             revert Module__ERC20PaymentClient__InvalidRecipient();
         }
     }
+
+    //--------------------------------------------------------------------------
+    // {ERC20PaymentClient} Function Implementations
+
+    /// @dev Ensures `amount` of payment tokens exist in address(this).
+    function _ensureTokenBalance(uint amount) internal virtual {
+        uint balance = __Module_orchestrator.token().balanceOf(address(this));
+
+        if (balance < amount) {
+            // Trigger callback from orchestrator to transfer tokens
+            // to address(this).
+            bool ok;
+            (ok, /*returnData*/ ) = __Module_orchestrator.executeTxFromModule(
+                address(__Module_orchestrator.fundingManager()),
+                abi.encodeWithSignature(
+                    "transferOrchestratorToken(address,uint256)",
+                    address(this),
+                    amount - balance
+                )
+            );
```

Comment:
Well, this is neat. But, there is an edge-case here.

Suppose we are dealing with tokens that take a fee on transfer ([examples](transferOrchestratorToken)) then even if you call the `transferOrchestratorToken` for `amount-balance` it wouldn't necessarily ensure that the contract has enough balance, because some fees will be deducted.

To remedy this, I recommend adding an assert statement that checks the contract balance again after the transfer, and only then you can *actually* be sure that the contract has enough balance.

Another remedy for this could be to simply create a whitelist of allowed proposal tokens and not allow any tokens that take a fee on transfer.